### PR TITLE
fix(aggs): make range/date_range upper bound `to` exclusive

### LINKED ISSUE #230: fix(aggs): range/date_range upper bound `to` is inclusive — adjacent ranges double-count boundary values
## Summary
`RangeCollector::collect` in the `range` aggregation compares values against the upper bound `to` with `<=` rather than `<`. This makes `to` inclusive, so documents whose value equals the boundary between two adjacent ranges are counted in **both** buckets — breaking the disjoint-bucket contract, inflating sub-aggregation populations, and diverging from Elasticsearch-compatible semantics. `date_range` delegates to the same collector and inherits the defect.

## Affected code
- `searchlite-core/src/query/aggs/mod.rs:1105` — `RangeCollector::collect` binds `lt_to = entry.to.map(|t| *val <= t).unwrap_or(true)`. The local is named **`lt_to`** ("less than to"), which is inconsistent with `<=`.
- `searchlite-core/src/query/aggs/mod.rs:1142-1205` — `DateRangeCollector::new` wraps a `RangeCollector`, so every `date_range` aggregation exhibits the same bug.

## Reproduction / evidence

Current code:
```rust
// searchlite-core/src/query/aggs/mod.rs:1102-1113
for entry in self.ranges.iter_mut() {
  if values.iter().any(|val| {
    let ge_from = entry.from.map(|f| *val >= f).unwrap_or(true);
    let lt_to = entry.to.map(|t| *val <= t).unwrap_or(true);  // <= should be <
    ge_from && lt_to
  }) {
    entry.bucket.doc_count += 1;
    ...
  }
}
```

The published Node example (`searchlite-node/README.md` under "Range Aggregation") demonstrates the canonical disjoint-range pattern:

```javascript
ranges: [
  { key: 'cheap',   to: 50 },
  { key: 'mid',     from: 50, to: 100 },
  { key: 'premium', from: 100 },
]
```

With the current code:
- `price = 50` → matches `cheap` (50 <= 50) **and** `mid` (50 >= 50 && 50 <= 100).
- `price = 100` → matches `mid` (100 <= 100) **and** `premium` (100 >= 100).

A failing assertion can be derived directly from the existing test at `searchlite-core/tests/aggregations.rs:2073` (`range_aggregation_counts`), which currently *encodes* the buggy behavior. Documents have `score ∈ {1, 5, 10, 20}`; ranges are `low: {to: 5}` and `mid: {from: 5, to: 15}`. The test expects `low = 2, mid = 2`, but summing them yields 4 across buckets even though only 3 docs fall into any range at all — the `score = 5` doc is counted in both `low` and `mid`. The test passes only because the same `<=` defect is baked into the expected values.

## Expected behavior
`from` inclusive, `to` exclusive (matching Elasticsearch's range aggregation and the disjoint-bucket intent of the Node README example). A value equal to a range's `to` lands in the next range up, never both. `bucket_count` values sum to the number of docs that fall into any range (i.e., each matching doc is counted at most once per top-level `range` agg).

## Actual behavior
`to` is inclusive. Documents on a bucket boundary are counted in two adjacent buckets; sub-aggregations under those buckets see inflated populations. `date_range` has the same behavior, including around exact ISO timestamps that correspond to `from`/`to` boundaries.

## Root cause (hypothesis)
Git history for this line:

- `1a35588 test: add aggregation coverage and benchmarks` introduced `let lt_to = entry.to.map(|t| val < t).unwrap_or(true);` — correct semantics.
- `f5e05ba feat: implemented aggregations` silently rewrote it to `val <= t`. The commit message does not mention a semantic change; the variable name was left as `lt_to`. The `=` appears to have been introduced during a larger reshuffle of this code, not as a deliberate API decision.

There is no documentation in the repo (neither `docs/aggregations.md`, `search-request.schema.json`, nor `openapi.yaml`) declaring `to` to be inclusive; the only "inclusive on both ends" statement in `docs/filters.md` refers to `I64Range`/`F64Range` **filters**, not to the range aggregation.

## Suggested fix
Restore `*val < t` at `searchlite-core/src/query/aggs/mod.rs:1105`, and update the expectations in `range_aggregation_counts` (`searchlite-core/tests/aggregations.rs:2073`) so the `score = 5` doc is counted in `mid` but not `low` (`low = 1, mid = 2`). No changes needed to `DateRangeCollector` — it inherits the fix.

## Environment
- Commit: `0967ca5` (HEAD of `main` at time of hunt)
- Rust MSRV in repo: 1.88.0
- Found by: scheduled Claude Code bug hunt

### PR DESCRIPTION
**Pull request by davidkelley**
## Summary

`RangeCollector::collect` in `searchlite-core/src/query/aggs/mod.rs` compared values against the upper bound `to` with `<=` rather than `<`. This made `to` inclusive, so documents whose value equalled the boundary between two adjacent ranges were counted in **both** buckets — breaking the disjoint-bucket contract, inflating sub-aggregation populations, and diverging from Elasticsearch-compatible `[from, to)` semantics. `date_range` delegates to the same collector and inherited the defect.

The collector now uses `*val < t`, restoring the **`from` inclusive, `to` exclusive** convention that the disjoint-range example in `searchlite-node/README.md` (`cheap` / `mid` / `premium`) already assumes and that Elasticsearch documents.

## What changed

- **`searchlite-core/src/query/aggs/mod.rs`** — one-character change: `*val <= t` → `*val < t` at the `RangeCollector::collect` call site. The local was already named `lt_to` ("less than to"), which was inconsistent with `<=`. `DateRangeCollector` inherits the fix automatically via its `RangeCollector` wrapper — no additional change needed.
- **`searchlite-core/tests/aggregations.rs`** — three test updates:
  - `range_aggregation_counts`: update expectations so the boundary value (`score = 5`) is counted once in `mid` and **not** also in `low` (now `low = 1, mid = 2` instead of the buggy `2, 2`).
  - New `range_aggregation_to_is_exclusive_at_boundary`: covers the full documented `cheap` / `mid` / `premium` disjoint-range layout with values placed precisely on every boundary (`25, 50, 75, 100, 150`), asserts per-bucket counts **and** that the total `doc_count` across buckets equals the number of matching docs (so no boundary double-counting is possible).
  - New `date_range_to_is_exclusive_at_boundary`: covers the delegated `date_range` path around an exact ISO boundary timestamp (`1970-01-01T00:00:02Z`), asserting the boundary doc lands in the upper range only.

## Test plan

- [x] `cargo test -p searchlite-core --test aggregations` — 40/40 pass (38 existing + 2 new regression tests, 1 expectation updated).
- [x] `cargo test -p searchlite-core` — full core test matrix green (466 tests across lib + all integration suites).
- [x] `cargo fmt --all --check` clean.
- [x] `cargo clippy -p searchlite-core --tests --no-deps -- -D warnings` clean.
- [x] Verified the new `range_aggregation_to_is_exclusive_at_boundary` test catches the original bug: under the old `<=` behaviour its `doc_count` sum exceeds 5 (boundary values are double-counted).

Closes #230

### CODE REVIEW

File: searchlite-core/tests/aggregations.rs [hunk 1]
```diff
@@ -3868,3 +3870,258 @@ fn pipeline_missing_metric_path_with_gap_policy_inserts_zeros() {
...
+    // premium: 100 and 150
+    assert_eq!(
+      buckets[2].doc_count, 2,
+      "premium should contain price=100 and price=150"
+    );
+    // Total across buckets must equal the number of matching docs (no double-counting).
+    let total: u64 = buckets.iter().map(|b| b.doc_count).sum();
+    assert_eq!(
+      total, 5,
+      "each doc must be counted exactly once across disjoint ranges"
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply searchlite-core/tests/aggregations.rs]: Good catch — applied the suggestion in f494c2e. The comment and assertion message now explicitly state the single-valued fixture assumption so readers don't infer an incorrect general contract for multi-valued fields.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by davidkelley** [comment]
@codex review

---
_Generated by [Claude Code](https://claude.ai/code)_
